### PR TITLE
nixos/rl-2211: make it clear NixOS isn't a set of packages

### PR DESCRIPTION
**Pull request by rapenne-s**
###### Description of changes

I think we should make it clear NixOS is a linux distribution, and not something you can use on Linux/MacOS because people may mix NixOS and Nix package manager.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/doc/manual/from_md/release-notes/rl-2211.section.xml [hunk 1]
```diff
@@ -2,8 +2,8 @@
   <title>Release 22.11 (“Raccoon”, 2022.11/30)</title>
   <para>
     The NixOS release team is happy to announce a new version of NixOS
-    22.11. NixOS is both a Linux distribution, and a set of packages
-    usable on other Linux systems and macOS.
+    22.11. NixOS is a Linux distribution, whose set of packages can be
+    used on other Linux systems and macOS.
```
Comment (fricklerhandwerk) [inline nixos/doc/manual/from_md/release-notes/rl-2211.section.xml:6]: What about making Nixpkgs explicit?
```suggestion
    22.11. NixOS is a Linux distribution based on Nixpkgs.
    Nixpkgs is a package collection which can be also used on other Linux systems and macOS.
```

**Review by fricklerhandwerk — APPROVED** [review]
Thanks, this is valuable. People confuse Nix, Nixpkgs and NixOS all the time.

There was a [request by the release team](https://github.com/NixOS/nix.dev/issues/296#issuecomment-1272630299) to collaborate with the @NixOS/documentation-team, and unfortunately we were simply too busy with other things to help out (me, specifically). 

I still think the docs team absolutely *should* help out in the future to avoid such retrospective changes, and we have yet to find the capacity to do as well as an effective and efficient process.

File: nixos/doc/manual/from_md/release-notes/rl-2211.section.xml (hunk 1 above)
Comment (winterqt) [reply nixos/doc/manual/from_md/release-notes/rl-2211.section.xml:6]: This is what I was leaning towards, I think it may be better.

**Comment by rapenne-s** [comment]
If you want to provide an alternative text, feel free to push changes to this PR or make another PR, I may not be active in the coming days.

File: nixos/doc/manual/release-notes/rl-2211.section.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Release 22.11 (“Raccoon”, 2022.11/30) {#sec-release-22.11}
 
-The NixOS release team is happy to announce a new version of NixOS 22.11. NixOS is both a Linux distribution, and a set of packages usable on other Linux systems and macOS.
+The NixOS release team is happy to announce a new version of NixOS 22.11. NixOS is a Linux distribution, whose set of packages can be used on other Linux systems and macOS.
```
Comment (mweinelt) [inline nixos/doc/manual/release-notes/rl-2211.section.md:3]:
```suggestion
The NixOS release team is happy to announce a new version of NixOS 22.11. NixOS is a Linux distribution, whose set of packages can also be used on other Linux systems and macOS.
```

**Review by mweinelt — APPROVED** [review]

**Comment by infinisil** [comment]
Ohh yeah this should go to master and be backported to 22.11. I'll change the base branch

**Comment by vcunat** [comment]
> but what specifically do you mean by retrospective changes here?

Probably that such things are better improved **before** the official release announcement.

**Review by vcunat — APPROVED** [review]

**Pull request merged**